google-cloud-firestore/src/main/java/com/google/cloud/firestore/AggregateQuery.java

```diff
@@ -185,6 +229,66 @@ private boolean shouldRetry(Throwable throwable) {
     public void onComplete() {}
   }
 
+  ApiFuture<QueryProfileInfo<AggregateQuerySnapshot>> getAggregateQueryProfileInfo(
+      QueryMode queryMode) {
+    RunAggregationQueryRequest request = toProto().toBuilder().setMode(queryMode).build();
+    final SettableApiFuture<QueryProfileInfo<AggregateQuerySnapshot>> result =
+        SettableApiFuture.create();
+
+    ResponseObserver<RunAggregationQueryResponse> observer =
+        new ResponseObserver<RunAggregationQueryResponse>() {
+          Timestamp readTime;
+          Map<String, Value> aggregateFieldsMap = new HashMap<>();
+          Struct planStruct = Struct.getDefaultInstance();
+          Struct statsStruct = Struct.getDefaultInstance();
+
+          @Override
+          public void onStart(StreamController streamController) {}
+
+          @Override
+          public void onResponse(RunAggregationQueryResponse response) {
+            if (readTime == null) {
+              readTime = Timestamp.fromProto(response.getReadTime());
+            }
+
+            if (aggregateFieldsMap.isEmpty() && response.hasResult()) {
+              aggregateFieldsMap =
+                  convertServerAggregateFieldsMapToClientAggregateFieldsMap(
+                      response.getResult().getAggregateFieldsMap());
+            }
+
+            if (response.hasStats()) {
+              if (response.getStats().hasQueryPlan()) {
+                planStruct = response.getStats().getQueryPlan().getPlanInfo();
+              }
+              if (response.getStats().hasQueryStats()) {
+                statsStruct = response.getStats().getQueryStats();
+              }
```

Comment:
nit: The same getter method is called multiple times. We should pull this out into a local variable.

```suggestion
              ResultSetStats stats = response.getStats();
              if (stats.hasQueryPlan()) {
                planStruct = stats.getQueryPlan().getPlanInfo();
              }
              if (stats.hasQueryStats()) {
                statsStruct = stats.getQueryStats();
              }
```